### ISSUE TITLE
UserObjectBindings::setUserAny and UserObjectBindings::getUserAny are marked as deprecated in the Ogre.i file

### ISSUE THREAD

**Issue by akacoding** [OGRECave/ogre#2180]
`UserObjectBindings::setUserAny` and `UserObjectBindings::getUserAny` are marked as deprecated in the Ogre.i file but are actually not deprecated if the UserObjectBindings header file is correct.

Furthermore the Any type does not deliver us the possiblities we need, to obtain an untyped `object` in C# (could be considered as `void*` in C++).

Do I miss something, or how can we use `UserObjectBindings` and `Any` with Swig?

**Comment by paroj** [comment]
> UserObjectBindings::setUserAny and UserObjectBindings::getUserAny are marked as deprecated in the Ogre.i file but are actually not deprecated if the UserObjectBindings header file is correct.

yes, this is wrong. It should only mark Renderable::getUserAny, not UserObjectBindings::getUserAny. I did not bother yet, as Any is not wrapped properly either.

To make Any work, one would need to
- instantiate the templated Any constructor for the types you need
- inject the casts you need into swig - similar to `castEntity()`

this does not give you a real untyped void*, but could get you somewhere if you use an IntPtr.
